consensus/system_contract/consensus.go

```diff
@@ -227,8 +227,31 @@ func (s *SystemContract) VerifyUncles(chain consensus.ChainReader, block *types.
 	return nil
 }
 
+// CalcBlocksPerSecond converts period to milliseconds and calculate blocks per second
+// For example: if Period = 250ms = 0.25s, then periodMs = 250
+func CalcBlocksPerSecond(periodMs uint64) uint64 {
+	blocksPerSecond := 1000 / periodMs // integer division, e.g. 1000/250 = 4
+	if blocksPerSecond == 0 {
+		blocksPerSecond = 1
+	}
+	return blocksPerSecond
+}
+
 func (s *SystemContract) CalcTimestamp(parent *types.Header) uint64 {
-	timestamp := parent.Time + s.config.Period
+	// Get the base timestamp (in seconds)
+	timestamp := parent.Time
```

Comment:
This change essentially means that we won't be able to support >1s block times, is that intentional?

Reply:
yeah I was thinking the same. maybe we should introduce an additional (optional) parameter that allows us to specify the amount of blocks if the period is 1. This way we can still support block times bigger than one and there's less risk when deploying the new version